crates/autopilot/src/infra/persistence/mod.rs

```diff
@@ -77,25 +87,31 @@ impl Persistence {
             .map_err(Error::DbError)
     }
 
-    /// Inserts the given events with the current timestamp into the DB.
-    /// If this function encounters an error it will only be printed. More
-    /// elaborate error handling is not necessary because this is just
-    /// debugging information.
-    pub fn store_order_events(&self, events: Vec<(domain::OrderUid, boundary::OrderEventLabel)>) {
+    /// Inserts an order event for each order uid in the given set.
+    /// Unique order uids are required to avoid inserting events with the same
+    /// label within the same order_uid. If this function encounters an error it
+    /// will only be printed. More elaborate error handling is not necessary
+    /// because this is just debugging information.
+    pub fn store_order_events(
+        &self,
+        order_uids: HashSet<domain::OrderUid>,
```

Comment:
I think there is no need for the `HashSet` anymore. IIRC basically all contexts where we store order events automatically have unique ids. Additionally we now always call the function that only inserts if we want to store a different label. Meaning that if we have duplicates we still only insert them once (albeit with slight overhead to determine it's a duplicate).
And finally if you think in terms of `DDD` the `domain` is only interested in passing the `infra` piece an `impl IntoIterator<Item = &OrderUid>` and the `infra` piece should ensure try to make sure every order is unique if this is even needed anymore.

Reply:
I started with the `IntoIterator` but ran into a problem using it together with the `async`. Replaced with `Vec` for now.